app/controllers/admin/videos_controller.rb

```diff
@@ -0,0 +1,45 @@
+class Admin::VideosController < ApplicationController
+  before_action :require_sign_in
+  before_action :require_admin
+
+  def new
+    @video = Video.new
+  end
+
+  def create
+    @video = Video.new(video_params)
+
+    handel_url
```

Comment:
This isn't a good place to handle such stuff. You modify the input in controller - this should be done in model if so. `before_create` callback and fix the url - however user inputs still should not be modified like that.
Proper way should be - save it to database as it was entered and validated and then whenever used - use a helper method in view.